packages/langium/test/grammar/type-system/type-validator.test.ts

```diff
@@ -497,4 +497,27 @@ describe('Property types validation takes in account types hierarchy', () => {
         });
     });
 
+    test('No false positive on declared type assignment', async () => {
+        const validation = await validate(`
+        interface A {}
+        interface B extends A {}
+        interface C extends B {}
+        interface D extends C {}
+        interface E extends C {}
+        interface F extends D {}
+        interface G extends D {}
+        interface H extends A {}
```

Comment:
Suggestion: I feel like there are a few interfaces in here which only add noise. We can probably just trim  of them down to `A, B, C` and accomplish the same effect.